src/main/scala/uk/gov/nationalarchives/omega/api/services/Dispatcher.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.api.services
+
+import cats.data.Validated.{ Invalid, Valid }
+import cats.data.{ Validated, ValidatedNec }
+import cats.effect.IO
+import cats.effect.std.Queue
+import com.fasterxml.uuid.{ EthernetAddress, Generators }
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import org.typelevel.log4cats.{ LoggerFactory, SelfAwareStructuredLogger }
+import uk.gov.nationalarchives.omega.api.business._
+import uk.gov.nationalarchives.omega.api.business.echo.{ EchoRequest, EchoService }
+import uk.gov.nationalarchives.omega.api.services.ServiceIdentifier.ECHO001
+
+import java.util.UUID
+
+class Dispatcher(val localProducer: LocalProducer, echoService: EchoService) {
+
+  implicit val loggerFactory: LoggerFactory[IO] = Slf4jFactory[IO]
+  implicit val logger: SelfAwareStructuredLogger[IO] = LoggerFactory[IO].getLogger
+
+  private val generator = Generators.timeBasedGenerator(EthernetAddress.fromInterface)
+
+  def run(dispatcherId: Int)(q: Queue[IO, LocalMessage]): IO[Unit] =
+    for {
+      requestMessage <- q.take
+      _ <- logger.info(s"Dispatcher # $dispatcherId, processing message id: ${requestMessage.persistentMessageId}")
+      (businessService, businessServiceRequest) <- IO.pure(createServiceRequest(requestMessage))
+      validatedBusinessServiceRequest <-
```

Comment:
Could we have a test scenario for an invalid request?

Reply:
I have added 3 new scenarios to `DispatcherSpec`. They all fail as we haven't implemented validation yet - this will be resolved in [PACT-836](https://national-archives.atlassian.net/browse/PACT-836).

Reply:
I'd say that it's better to mark them `pending` then have them fail.

Reply:
Confirmed.